src/storage/mod.rs

```diff
@@ -0,0 +1,98 @@
+use thiserror::Error;
+
+mod nats_kv;
+mod state;
+
+use semver::Version;
+use async_trait::async_trait;
+
+use state::{
+    LatticeState,
+    LatticeStateBuilder,
+    LatticeId,
+};
+
+/////////////
+// Engines //
+/////////////
+
+pub type EngineName = String;
+
+/// An engine for performing some task that is pluggable
+pub trait Engine {
+    /// Get the name of the lattice storage
+    fn name() -> EngineName;
```

Comment:
One more idea I just had: maybe make these pub consts rather than functions. See what I did with events

Reply:
my note above about being able to use a `RemoteStorageEngine` -- I want to be able to ask a storage engine it's name (and it possibly do some work that I don't know about to find out). I didn't go as far as making this `async name()`, but I kind of want to

Reply:
Resolving this due to the other discussion, slightly overkill, but enables some cool future use cases IMO!